### PR TITLE
Java page

### PR DESCRIPTION
**Pull request by alex-tpom6oh**

### CODE REVIEW

**Comment by hugihlynsson** [comment]
#### 📝Changed routes:


| [**…ntation/reference/java**](https://docs-336gwhgp4.avo-dev.app/implementation/reference/java) | [**…n/reference/javascript**](https://docs-336gwhgp4.avo-dev.app/implementation/reference/javascript) |
|:-:|:-:|
|<a href="https://docs-336gwhgp4.avo-dev.app/implementation/reference/java">Screenshot of /implementation/reference/java</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/wvkM3vQPEaLyZL2UVw9sVJXsbj_z9YIZFV-hcEytLPuOW7c6Upm2jfuZULNn8DPfTPIp0Jh33v6iXDaVlHatEw.png">(view full size)</a> |<a href="https://docs-336gwhgp4.avo-dev.app/implementation/reference/javascript">Screenshot of /implementation/reference/javascript</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/xLcxmoiekr0oJVGbEoGJMNtn7EHQCIe-ng17heuQsOuqjqsYyL2kA01BcaM_pZ5MtAZaBtKB7OKYuNLJ7be8Ag.png">(view full size)</a> |



| [**…ation/reference/kotlin**](https://docs-336gwhgp4.avo-dev.app/implementation/reference/kotlin) | [**…tation/reference/swift**](https://docs-336gwhgp4.avo-dev.app/implementation/reference/swift) |
|:-:|:-:|
|<a href="https://docs-336gwhgp4.avo-dev.app/implementation/reference/kotlin">Screenshot of /implementation/reference/kotlin</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/7Hz6qlJlpQysa35dpk2H8I4hbhb2RBhAA4Zq7nzcPTPMRQs_q00FPNKK1lcn05gbhj4ik8GJ32cuLIy6rc4A8Q.png">(view full size)</a> |<a href="https://docs-336gwhgp4.avo-dev.app/implementation/reference/swift">Screenshot of /implementation/reference/swift</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Z8ZITAumjUr4slQxJFpPmV6cfC4gdXpShUqtF45Io0uzFIvLj2AXROQZ2fuduCWWFmfF-KWW_b0RKJTmtAJXFA.png">(view full size)</a> |



| [**…-programming-languages**](https://docs-336gwhgp4.avo-dev.app/implementation/supported-programming-languages) |
|:-:|
|<a href="https://docs-336gwhgp4.avo-dev.app/implementation/supported-programming-languages">Screenshot of /implementation/supported-programming-languages</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/bqX9wxl5e1TUbEtn3CaLFx5SHHBQtFr_xGz2kv-9uE_gvX2o7bL3Tu-EqicsakqQ8zEvqUhgVOsqXq--VANAVA.png">(view full size)</a> |







Commit 6ac8a113b0c69d16319e4a796b1fbcaeeff780db (https://docs-336gwhgp4.avo-dev.app).

File: pages/implementation/reference/java.mdx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+Use the <Link passHref href="/workspace/tracking-plan/implementation-status"><a>Implementation status</a></Link> in your Avo workspace and the <Link passHref href="/implementation/inspector/sdk/android"><a>Avo Inspector</a></Link> to verify that your implementation is correct. If you don't want to use the Avo Inspector you can use the <a href="https://github.com/avohq/android-analytics-debugger">standalone visual debugger</a>.
+
+### Reference
+
+#### initAvo
+
+```java
+public static void initAvo(AvoEnv env,
+    // Other parameters may not be present, depending on your tracking plan
+    Integer systemProprty0, Boolean systemProperty1,
```
Comment (bragur) [inline pages/implementation/reference/java.mdx:77]:
```suggestion
    Integer systemProperty0, Boolean systemProperty1,
```

File: pages/implementation/reference/java.mdx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    ICustomDestination customDestDestination,
+    boolean strict = true, boolean noop = false,
+    application: Application, context: Context
+    );
+```
+
+There is also an init method that accepts an instance of the Avo Inpsector, use it if you have Avo Inspector set up in your project, so Avo Functions will automatically report invocations to the Avo Inspector.
```
Comment (bragur) [inline pages/implementation/reference/java.mdx:84]:
```suggestion
There is also an init method that accepts an instance of the Avo Inspector, use it if you have Avo Inspector set up in your project, so Avo Functions will automatically report invocations to the Avo Inspector.
```

File: pages/implementation/reference/java.mdx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+```java
+public static void initAvoWithInspector(AvoEnv env,
+    Object avoInspector,
+    // Other parameters may not be present, depending on your tracking plan
+    // Same parameters, as in `initAvo`
+    Integer systemProprty0, Boolean systemProperty1,
```
Comment (bragur) [inline pages/implementation/reference/java.mdx:91]:
```suggestion
    Integer systemProperty0, Boolean systemProperty1,
```

File: pages/implementation/reference/java.mdx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    ICustomDestination customDestDestination,
```
Comment (bragur) [inline pages/implementation/reference/java.mdx:92]: should this not be `customDestination`?

File: pages/implementation/reference/java.mdx [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+### Reference
+
+#### initAvo
+
+```java
+public static void initAvo(AvoEnv env,
+    // Other parameters may not be present, depending on your tracking plan
+    Integer systemProprty0, Boolean systemProperty1,
+    ICustomDestination customDestDestination,
```
Comment (bragur) [inline pages/implementation/reference/java.mdx:78]: `customDestination`?

File: pages/implementation/reference/java.mdx [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+`Object avoInspector`: optional Avo Inspector instance. If you use Avo Inspector pass it here to make the Avo functions automatically report the invocations to the Avo Inspector.
+
+###### Additional arguments
+
+`Application application` and `Context context` are required by some analytics destinations.
+
+#### setSystemProperties
+
+```java
+public static void setSystemProperties(Integer systemProprty0, Boolean systemProperty1, ...)
```
Comment (bragur) [inline pages/implementation/reference/java.mdx:157]:
```suggestion
public static void setSystemProperties(Integer systemProperty0, Boolean systemProperty1, ...)
```

**Review by bragur — APPROVED** [review]
LGTM, found a few typos

File: pages/implementation/reference/java.mdx (hunk 2 above)
Comment (alex-tpom6oh) [reply pages/implementation/reference/java.mdx:84]: Ah, this should be caught by the spellcheck script, there is a problem with it, will fix it here!

File: pages/implementation/reference/java.mdx (hunk 3 above)
Comment (alex-tpom6oh) [reply pages/implementation/reference/java.mdx:91]: This typo is all over the reference, good catch 😄

**Pull request merged**